### PR TITLE
add exact version info (build time / build id) to about window #618

### PR DESCRIPTION
**Pull request by poormusic2001**
as title

1. Make version string in settingsViewController copyable (User can also copy in the about window of catalyst version).
2. Build a new about window in the catalyst version.

### CODE REVIEW

File: Monal/Classes/HelperTools.m [hunk 1]
```diff
@@ -719,4 +719,17 @@ +(CIImage*) createQRCodeFromString:(NSString*) input
     return qrCode.outputImage;
 }
 
++(NSString*)appBuildVersionInfo
+{
+    NSString *versionTxt = @"";
+    NSDictionary* infoDict = [[NSBundle mainBundle] infoDictionary];
+    #if IS_ALPHA
+        versionTxt = [NSString stringWithFormat:@"Alpha %@ (%s: %s UTC)", [infoDict objectForKey:@"CFBundleShortVersionString"], __DATE__, __TIME__];
```
Comment (FriedrichAltheide) [inline Monal/Classes/HelperTools.m]: Please declare versionTxt here

File: Monal/Classes/HelperTools.m [hunk 2]
```diff
@@ -719,4 +719,17 @@ +(CIImage*) createQRCodeFromString:(NSString*) input
...
+    #else
+        versionTxt = [NSString stringWithFormat:@"%@ (%@)", [infoDict objectForKey:@"CFBundleShortVersionString"], [infoDict objectForKey:@"CFBundleVersion"]];
```
Comment (FriedrichAltheide) [inline Monal/Classes/HelperTools.m]: and here

File: Monal/Classes/MLSettingsTableViewController.m [hunk 3]
```diff
@@ -137,13 +138,8 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
...
             {
                 NSString* versionTxt = nil;
-                NSDictionary* infoDict = [[NSBundle mainBundle] infoDictionary];
-#if IS_ALPHA
-                versionTxt = [NSString stringWithFormat:@"Alpha %@ (%s: %s UTC)", [infoDict objectForKey:@"CFBundleShortVersionString"], __DATE__, __TIME__];
-#else
-                versionTxt = [NSString stringWithFormat:@"%@ (%@)", [infoDict objectForKey:@"CFBundleShortVersionString"], [infoDict objectForKey:@"CFBundleVersion"]];
-#endif
-                [cell initCell:@"Version" withLabel:versionTxt];
+                versionTxt = [HelperTools appBuildVersionInfo];
```
Comment (FriedrichAltheide) [inline Monal/Classes/MLSettingsTableViewController.m]: `NSString* versionTxt = [...` use one line

File: Monal/Classes/MLSettingsTableViewController.m [hunk 4]
```diff
@@ -240,6 +236,10 @@ -(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *
                     break;
                     
                 case 5:
+                {
+                    UIPasteboard *pastboard = UIPasteboard.generalPasteboard;
```
Comment (FriedrichAltheide) [inline Monal/Classes/MLSettingsTableViewController.m]: code style

File: Monal/Classes/MLSettingsAboutViewController.m [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+@property (weak, nonatomic) IBOutlet UITextView *aboutVersion;
+
+@end
+
+@implementation MLSettingsAboutViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    NSString *versionTxt = [HelperTools appBuildVersionInfo];
```
Comment (FriedrichAltheide) [inline Monal/Classes/MLSettingsAboutViewController.m]: code style

File: Monal/Classes/MLSettingsAboutViewController.m [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    [self.aboutVersion setText:versionTxt];
+    
+    UITapGestureRecognizer *tapReCognizer = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(copyTxt:)];
```
Comment (FriedrichAltheide) [inline Monal/Classes/MLSettingsAboutViewController.m]: code style

File: Monal/Classes/MonalAppDelegate.m [hunk 7]
```diff
@@ -630,17 +631,30 @@ - (void)buildMenuWithBuilder:(id<UIMenuBuilder>)builder
         
         UIMenu *detailsMenu = [UIMenu menuWithTitle:@"" image:nil identifier:@"im.monal.detail" options:UIMenuOptionsDisplayInline children:@[detailsCommand]];
         [builder insertSiblingMenu:detailsMenu afterMenuForIdentifier:@"im.monal.new"];
-        
-        
-       UIKeyCommand *deleteCommand = [UIKeyCommand commandWithTitle:@"Delete Conversation" image:nil action:@selector(deleteConversation) input:@"\b" modifierFlags:UIKeyModifierCommand propertyList:nil];
+                
+        UIKeyCommand *deleteCommand = [UIKeyCommand commandWithTitle:@"Delete Conversation" image:nil action:@selector(deleteConversation) input:@"\b" modifierFlags:UIKeyModifierCommand propertyList:nil];
```
Comment (FriedrichAltheide) [inline Monal/Classes/MonalAppDelegate.m]: code style

File: Monal/Classes/MonalAppDelegate.m [hunk 8]
```diff
@@ -630,17 +631,30 @@ - (void)buildMenuWithBuilder:(id<UIMenuBuilder>)builder
...
+        [builder replaceChildrenOfMenuForIdentifier:UIMenuAbout fromChildrenBlock:^NSArray<UIMenuElement *> * _Nonnull(NSArray<UIMenuElement *> * _Nonnull items) {
+            UICommand *itemCommand = (UICommand*)items.firstObject;
+            UICommand *aboutCommand = [UICommand commandWithTitle:itemCommand.title image:nil action:@selector(aboutWindow) propertyList:nil];
+            NSArray *menuItems = @[aboutCommand];
+            return menuItems;
+        }];
     }
 }
 
+-(void) aboutWindow {
```
Comment (FriedrichAltheide) [inline Monal/Classes/MonalAppDelegate.m]: code style

**Review by FriedrichAltheide — CHANGES_REQUESTED** [review]

File: Monal/Classes/HelperTools.h [hunk 9]
```diff
@@ -70,6 +70,8 @@ void logException(NSException* exception);
 +(monal_void_block_t) startTimer:(double) timeout withHandler:(monal_void_block_t) handler andFile:(char*) file andLine:(int) line andFunc:(char*) func;
 +(monal_void_block_t) startTimer:(double) timeout withHandler:(monal_void_block_t) handler andCancelHandler:(monal_void_block_t _Nullable) cancelHandler andFile:(char*) file andLine:(int) line andFunc:(char*) func;
 
++(NSString*)appBuildVersionInfo;
```
Comment (FriedrichAltheide) [inline Monal/Classes/HelperTools.h]: code style

File: Monal/Classes/MLSettingsAboutViewController.m [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+@implementation MLSettingsAboutViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    NSString* versionTxt = [HelperTools appBuildVersionInfo];
+    [self.aboutVersion setText:versionTxt];
+    
+    UITapGestureRecognizer* tapReCognizer = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(copyTxt:)];
```
Comment (FriedrichAltheide) [inline Monal/Classes/MLSettingsAboutViewController.m:25]: Why don't you setup the tap gesture for the version label in the storyboard?

File: Monal/Classes/MLSettingsAboutViewController.m [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+    [self.aboutVersion addGestureRecognizer:tapReCognizer];
+    UIBarButtonItem* rightBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemClose target:self action:@selector(close:)];
+    self.navigationItem.rightBarButtonItem = rightBarButtonItem;
+}
+
+- (NSArray<UIKeyCommand *>*)keyCommands {
```
Comment (FriedrichAltheide) [inline Monal/Classes/MLSettingsAboutViewController.m]: code style

**Review by FriedrichAltheide — CHANGES_REQUESTED** [review]

File: Monal/Classes/MLSettingsAboutViewController.m (hunk 10 above)
Comment (poormusic2001) [reply Monal/Classes/MLSettingsAboutViewController.m:25]: 1. I prefer to setup tap gesture in code.
2. Put many things in storyboard will load a very long time in my MacBook Air, like the Main.Storyboard.

So, I setup the tap gesture for the version label in the code. You can change it to the storyboard if you insist.

**Pull request merged**